### PR TITLE
Adaptation of SDP to incorporate inequality constraints.

### PR DESCRIPTION
**Pull request by manish7294**
Included sparseInequalityA and denseInequalityA and have made required changes to LRSDP, primal-dual and matrix completion method.

### CODE REVIEW

**Comment by manish7294** [comment]
@rcurtin I think it's ready for review now. Though I am facing a problem with GaussianMatrixSensingSDP, I am not getting why it is failing in python2 instances while as passing in python3 instance. I am also not able to reproduce it as I have python3 install on my system. If you see anything wrong then please point it to me. Sorry for asking all the little things. Will try to setup a python2 VM on AWS :)

**Comment by manish7294** [comment]
Yeah! I agree with you. We can close this as of now, I think we can always open it if it will be needed in future by anyone :)

File: src/mlpack/core/optimizers/sdp/primal_dual.hpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ class PrimalDualSolver
   //! Starting lagrange multiplier for the sparse constraints.
   arma::vec initialYsparse;
 
-  //! Starting lagrange multiplier for the sparse constraints.
+  //! Starting lagrange multiplier for the dense constraints.
```
[the remark this answers is not in the document]
Comment (manish7294) [reply src/mlpack/core/optimizers/sdp/primal_dual.hpp]: :)

**Comment by rcurtin** [comment]
I grabbed the spelling fix in 949ab83ef, and I'll go ahead and close this then. :+1:

**Pull request closed without merging**